package-lock.json

```diff
@@ -1548,6 +1548,21 @@
         "@types/jsonld": "^1.5.0",
         "json2typescript": "1.4.1",
         "jsonld": "^1.8.0"
+      },
+      "dependencies": {
+        "json2typescript": {
+          "version": "1.4.1",
+          "resolved": "https://registry.npmjs.org/json2typescript/-/json2typescript-1.4.1.tgz",
+          "integrity": "sha512-l+gmzjK4klnIG4Nr9SdHlhc03e63ln6UN1AASTTufrRoG3MjR443HqGmx/1OMOb/xvp7Ztb7UKZIzxXyDjGSsw=="
+        },
+        "rxjs": {
```

Comment:
@kilchenmann This wrong because it will install a private dependency of `rxjs` for @knora/api. It should be removed.

Reply:
Please delete `package-lock.json` and remove your local `node_modules` and run `npm i`. Then check `package-lock.json` again.

Reply:
Angular installs `rxjs` ~6.5.5.

Reply:
@tobiasschweizer Thank you. This is exactly what I did, but I will try again...

Reply:
please see https://github.com/dasch-swiss/knora-api-js-lib/blob/master/README.md#dependencies-and-peer-dependencies

Reply:
It was the following problem: I had to reset package-lock.json because of 180 merge conflicts in this file. To make the reset, I deleted the folder `node_modules` and `package-lock.json`. Then I re-run `npm i`. But the problem was, that I already added @knora/api with `yalc`, which was then ready to install from `.yalc`-folder. This added some "unnecessary" dependencies to `package-lock.json` in the section `@knora/api`.

The solution to reset package-lock.json should be as follow:

```terminal
rm -rf node_modules
rm -rf .yalc
rm package-lock.json
npm i
yalc add @knora/api
```

Reply:
Thank you @tobiasschweizer for the hints.

Reply:
I think you can also use `yalc remove --all`

Reply:
after `yalc add @knora/api` you should run `npm i` again.

Reply:
> But the problem was, that I already added @knora/api with `yalc`, which was then ready to install from `.yalc`-folder.

maybe it was still an older version that had a dependency for `rxjs` and not a peer dependency as the newer version.

In any case we need to document that in the README. We could do it together if you like.

Reply:
> after yalc add @knora/api you should run npm i again.

Yes, of course. But if package-lock.json is missing, it will add the dependencies which gives the conflict.

Reply:
> maybe it was still an older version that had a dependency for rxjs and not a peer dependency as the newer version.

I don't think so. It was the fresh builded version from Yesterday evening.

Reply:
I will checkout your branch and try it from scratch.

Reply:
I could rebuild everything from scratch and it works for me. However, we need to define a step-by-step guideline. Could you make a user story for this?

I could, however, provoke the issue locally by manipulating `.yalc/@knora/api/package.json`:

```
"dependencies": {
    "@types/jsonld": "^1.5.0",
    "json2typescript": "1.4.1",
    "jsonld": "^1.8.0",
    "rxjs": "~6.4.0"
  }
```

Now this behaves like the old version which had no peer deps when building the lib:

> ERROR: projects/knora-ui/src/lib/viewer/operations/display-edit/display-edit.component.ts:97:9 - error TS2345: Argument of type 'OperatorFunction<WriteValueResponse, ApiResponseError | ReadResource>' is not assignable to parameter of type 'OperatorFunction<ApiResponseError | WriteValueResponse, ApiResponseError | ReadResource>'.
  Types of parameters 'source' and 'source' are incompatible.
    Type 'Observable<ApiResponseError | WriteValueResponse>' is not assignable to type 'Observable<WriteValueResponse>'.
      The types of 'source.operator.call' are incompatible between these types.
        Type '(subscriber: import("/.../knora-ui-ng-lib/node_modules/@knora/api/node_modules/rxjs/internal/Subscriber").Subscriber<any>, source: any) => import("/.../gitreps/knora-ui-ng-lib/node_modules/@knora/api/node_modules/rxjs/internal/types").TeardownLogic' is not assignable to type '(subscriber: import("/Users/.../knora-ui-ng-lib/node_modules/rxjs/internal/Subscriber").Subscriber<any>, source: any) => import("/.../knora-ui-ng-lib/node_modules/rxjs/internal/types").TeardownLogic'.
          Types of parameters 'subscriber' and 'subscriber' are incompatible.
            Type 'Subscriber<any>' is missing the following properties from type 'Subscriber<any>': _parent, _parents, _addParent

> 97         mergeMap((res: WriteValueResponse) => {
>           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 98           return this.knoraApiConnection.v2.values.getValue(this.parentResource.id, res.uuid);
>   > ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 99         })
>   ~~~~~~~~~~

This is because the `Observable` from two different versions of the same lib is not considered to be the same class.

Reply:
> However, we need to define a step-by-step guideline. Could you make a user story for this?

Done: https://dasch.myjetbrains.com/youtrack/issue/DSP-199